Extension/lib/content-script/assistant/js/adguard-selector.js

```diff
@@ -417,44 +481,87 @@ var AdguardSelectorLib = (function (api, $) {
         return false;
     };
 
+    /********** iOS modifications ***************/
+    var iosElementSelectHandler = function (e) {
+
+        sgMouseoverHandler.call(this, e);
+        sgMousedownHandler.call(this, e);
+    };
+
+    var needIgnoreTouchEvent = function () {
+
+        if (ignoreTouchEvent > 0) {
+
+            ignoreTouchEvent--;
+            return true;
+        }
+
+        return false;
+    };
+
+    var iosElementTouchendHandler = function (e) {
+        e.stopPropagation();
+
+        if (needIgnoreTouchEvent()) {
+            return true;
+        }
+
+        iosElementSelectHandler.call(this, e);
+        return false;
+    };
+
+    var iosPreventEventHandler = function (e) {
+        e.stopPropagation();
+        // e.preventDefault();
+
+        return false;
+    };
+
+    var iosGestureEndHandler = function (e) {
+        ignoreTouchEvent = 2;
+        return true;
+    };
+
+    var iosTouchMoveHandler = function (e) {
+        ignoreTouchEvent = 1;
+        return true;
+    };
+
+    /*******************************************/
 
     var setupEventHandlers = function () {
         makeIFrameAndEmbededSelector();
 
         var sgIgnore = $("body *:not(." + IGNORED_CLASS + ")");
-        sgIgnore.on("mouseover", sgMouseoverHandler);
-        sgIgnore.on("mouseout", sgMouseoutHandler);
-        sgIgnore.on("click", sgMousedownHandler);
+
+        if (isIOS) {
```

Comment:
@Stillness-2 is it possible to do it without this flag?
We can do it more generic, touch events could be useful not only for ios.
Or common events(mouseover, mouseout, click) don't work in ios?